### PR TITLE
WRC - Node - Handling errors 

### PR DESCRIPTION
**Pull request by mihaiberq**
Errors:
  [x] is this [insert example] an operational or programmer error?
  [x] how do you handle operational errors?
  [x] how do you handle programmer errors?

### CODE REVIEW

**Comment by enki-hq** [comment]
- [ ] https://enkipro.com/preview-insight/enkidevs/curriculum/javascript/node/operational-vs-programmer-errors/handling-operational-errors.md?ref=e22443efb3bc749ead78fed7c53db3ae3615f04a
- [ ] https://enkipro.com/preview-insight/enkidevs/curriculum/javascript/node/operational-vs-programmer-errors/handling-programmer-errors.md?ref=e22443efb3bc749ead78fed7c53db3ae3615f04a
- [ ] https://enkipro.com/preview-insight/enkidevs/curriculum/javascript/node/operational-vs-programmer-errors/operational-vs-programmer-errors.md?ref=e22443efb3bc749ead78fed7c53db3ae3615f04a

File: javascript/node/operational-vs-programmer-errors/handling-operational-errors.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+For most operational errors, the message will most likely point to what is left to do. For example, trying to access a file that is not yet created will *return* (async) or *throw* (sync) an `ENOENT: no such file or directory` error.
+
+While we're at it, let's focus on the difference between asynchronous and synchronous functions. Consider the following snippet:
+```js
+import fs from 'fs';
+
+function readAndParseFile(path,
+  user_callback) {
+    try {
+      fs.read(path, (err, text) => {
```
Comment (lizTheDeveloper) [inline javascript/node/operational-vs-programmer-errors/handling-operational-errors.md]: `fs.read` takes a file descriptor, which is an integer, you are likely thinking of `fs.readFile` here.

File: javascript/node/error-handling/README.md [hunk 2]
```diff
@@ -2,10 +2,12 @@ name: Error Handling
...
 type: insights-list
 
-description: How to handle various errors. 
+description: How to handle various errors.
 
 section: 1
 
+parent: operational-vs-programmer-errors
+
 insights:
```
Comment (lizTheDeveloper) [inline javascript/node/error-handling/README.md]: Let's actually just merge these workouts, call it error handling. Use this ordering:
- operational-vs-programmer-errors
- handling-operational-errors
- handling-programmer-errors
- try-catch-only-for-sync-code
- first-error-callbacks-in-node

File: javascript/node/operational-vs-programmer-errors/handling-programmer-errors.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# Handling Operational Errors
```
Comment (lizTheDeveloper) [inline javascript/node/operational-vs-programmer-errors/handling-programmer-errors.md]: Fix title, handling programmer errors

File: javascript/node/operational-vs-programmer-errors/operational-vs-programmer-errors.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+* Unhandled operational error
+* Handled programmer error
+* Unhandled programmer error
+
+---
+## Revision
+
+A programmer error is
+
+???
```
Comment (lizTheDeveloper) [inline javascript/node/operational-vs-programmer-errors/operational-vs-programmer-errors.md]: Use these answers, I think they trigger more considered thinking
"a problem where the program unexpectedly crashes"
"a problem where the program expectedly crashes"
"a problem with the server configuration"
"a problem on the client"
"an unhandled exception"

**Comment by lizTheDeveloper** [comment]
merged in #494

**Pull request closed without merging**